aidb/inference/bound_inference_service.py

```diff
@@ -194,41 +195,16 @@ async def infer(self, inputs: pd.DataFrame):
 
     # Note: the input columns are assumed to be in order
     async with self._engine.begin() as conn:
-      is_in_cache = await self._check_inputs_in_cache_table(inputs, conn)
-      results = []
+      out_cache_inputs = await self._get_inputs_not_in_cache_table(inputs, conn)
       records_to_insert_in_table = []
-      inputs_to_insert_in_cache_table = []
-      # TODO
-      # - Batch the service inference
-      # - Batch the selection for cached results... How to do this?
-      for idx, (_, inp_row) in self.optional_tqdm(enumerate(inputs.iterrows()), total=len(inputs)):
-        if is_in_cache[idx]:
-          query = self._result_query_stub.where(
-            sqlalchemy.sql.and_(
-              *[getattr(self._cache_table.c, self.convert_normalized_col_name_to_cache_col_name(col)) == pandas_dtype_to_native_type(getattr(inp_row, col)) for col in
-                self.binding.input_columns]
-            )
-          )
-          df = await conn.run_sync(lambda conn: pd.read_sql(query, conn))
-          results.append(df)
-        else:
-          inference_results = self.service.infer_one(inp_row)
-          for idx, col in enumerate(self.binding.output_columns):
-            if idx < len(inference_results.columns):
-              inference_results.rename(columns={inference_results.columns[idx]: col}, inplace=True)
-            else:  # assume later columns are copied from input
-              # assume columns with same name in input / output tables are the same
-              _, col_n = col.split('.')
-              for c in self.binding.input_columns:
-                if c.split('.')[1] == col_n:
-                  inference_results[col] = inp_row[c]
-                  break
-          records_to_insert_in_table.append(inference_results)
-          results.append(inference_results)
-          inputs_to_insert_in_cache_table.append(inp_row)
-      await self._insert_in_cache_table(inputs_to_insert_in_cache_table, conn)
+      bs = self.service.preferred_batch_size
+      input_batches = [out_cache_inputs.iloc[i:i + bs] for i in range(0, len(out_cache_inputs), bs)]
```

Comment:
Where did the caching logic move to? It looks like a bunch of code got deleted (which could be okay if some simplifying assumptions were made). But I want to make sure we didn't delete some of the important logic

Reply:
It’s been moved to below. The actual deleted part is to query result table and append to results list, which is a duplicate because we will not use this results list in any way.